generic3g/vertical/ModelVerticalGrid.F90

```diff
@@ -120,16 +120,28 @@ subroutine get_coordinate_field(this, field, coupler, standard_name, geom, typek
        integer, optional, intent(out) :: rc
 
        integer :: status
+       character(len=ESMF_MAXSTR) :: short_name
        type(VirtualConnectionPt) :: v_pt
        type(StateItemExtension), pointer :: new_extension
        class(StateItemSpec), pointer :: new_spec
        type(FieldSpec) :: goal_spec
+       type(VerticalDimSpec) :: vertical_dim_spec
        integer :: i
 
-       v_pt = VirtualConnectionPt(state_intent='export', short_name=this%variants%of(1))
-       goal_spec = FieldSpec(geom=geom, vertical_grid=this, vertical_dim_spec=VERTICAL_DIM_EDGE, &
-            typekind=typekind, standard_name=standard_name, units=units, &
-            ungridded_dims=UngriddedDims())
+       short_name = this%variants%of(1)
+       v_pt = VirtualConnectionPt(state_intent='export', short_name=short_name)
+       select case (short_name)
+       case ("PLE")
+          vertical_dim_spec = VERTICAL_DIM_EDGE
+       case ("PL")
+          vertical_dim_spec = VERTICAL_DIM_CENTER
+       case default
+          _FAIL("short name should be one of PL/PLE, not" // trim(short_name))
+       end select
+
```

Comment:
Ah - ok - this needs to be generalized.  It can go in now, but please open a ticket to generalize and that will be your next priority after we merge this PR.

The idea is that someone else might provide a dyncore that uses a different short name and we don't want to be tied to FV's decision here.   Instead we rely on (a) standard name and (b) staggering.   As we register fields as coodinates we specify the standard name and the staggering.    The "get" procedure then uses staggering and standard name to "find" the appropriate coordinates.

In theory this involves creating a gFTL map with key: {string, vert stagger}. A bit heavyweight, but there's just enough going on here that you probably don't want to reinvent the wheel.

Reply:
> Instead we rely on (a) standard name and (b) staggering. As we register fields as coodinates we specify the standard name and the staggering.I'll address that.

One concern here is that `standard_name` is optional in FieldSpec and also VariableSpec. Is there someplace where standard name is enforced?

Reply:
It is not optional for a vertical grid. Note that run-of-the-mill fields do not need to specify a standard name, but FixedLevels absolutely must indicate whether to use PLE vs ZLE.    My thinking is that standard_name will fill that role nicely, but am open to other solutions on that front.

Enforcing that standard name is specified for things in a vertical grid spec should be added in due course.

Reply:
> It is not optional for a vertical grid. 

I mean not optional in the sense that we need it.   Might be optional in the Fortran sense, in which case we check.